Base.rte/Effects/Pyro/Flame/Flame.lua

```diff
@@ -0,0 +1,52 @@
+function Create(self)
+	self.strength = 80;	-- Damage frequency factor
+	self.checkTimer = Timer();
+	self.checkDelay = math.random(20);
+end
+
+function Update(self)
+	self.ToSettle = false;
+	if self.target and IsMOSRotating(self.target) then
+		self.Vel = self.target.Vel;
+		self.Pos = self.target.Pos - Vector(self.stickPos.X, self.stickPos.Y):RadRotate(self.target.RotAngle - self.tsAngle);
+		local actor = MovableMan:GetMOFromID(self.target.RootID);	-- Inflict damage on the root actor
+		if MovableMan:IsActor(actor) then
+			actor = ToActor(actor);
+			if math.random() < (self.strength * self.target.DamageMultiplier) / actor.Mass / (self.target.Material.StructuralIntegrity) then
+				actor.Health = actor.Health - 1;
+			end
+		end
+	else
+		self.target = nil;
+		if self.checkTimer:IsPastSimMS(self.checkDelay) then
+			self.checkTimer:Reset();
+			self.checkDelay = self.checkDelay + 1;	-- Gradually extend the delay for optimization reasons
+			local checkPos = self.Pos + Vector(self.Vel.X, self.Vel.Y):SetMagnitude(math.sqrt(self.Vel.Magnitude));
+			local mocheck = SceneMan:GetMOIDPixel(checkPos.X, checkPos.Y);
+			if mocheck ~= 255 then
+				local mo = MovableMan:GetMOFromID(mocheck);
+				if mo then
+					self.target = ToMOSRotating(mo);
+					self.tsAngle = mo.RotAngle;	-- Target angle when sticking	
+					self.stickPos = SceneMan:ShortestDistance(mo.Pos, self.Pos, SceneMan.SceneWrapsX) * 0.8;
+					self.Pos = mo.Pos - Vector(self.stickPos.X, self.stickPos.Y);
+				end
+			end
+		end
+	end
+	local age = self.Age * 0.0001 + 1;
```

Comment:
This is a place I would put a comment, not explaining what the variable is, but what this set of code is for.
Why are you comparing chance with age in seconds + 1, i.e. what does this section of code actually do?